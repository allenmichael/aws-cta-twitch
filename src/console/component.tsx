import * as React from 'react';
import './component.sass';

interface ConsoleEntry {
  frame: string;
  log: string;
}

interface State {
  logHistory: ConsoleEntry[];
}

export class Console extends React.Component<{}, State> {
  public state: State = {
    logHistory: window.rig.history || []
  }

  private updateConsole() {
    this.setState({
      logHistory: window.rig.history
    });
  }

  public componentDidMount() {
    window.rig.update = () => {
      this.updateConsole();
    }

    this.updateConsole();
  }

  public render() {
    const logs = this.state.logHistory.map((entry, index) => (
      <div key={index}>{entry.frame} $ {entry.log}</div>
    ));

    return (
      <div className="console">{logs}</div>
    );
  }
}
