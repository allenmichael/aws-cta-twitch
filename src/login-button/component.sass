.login-button
  float: right
  color: white
  background-color: #7D55C7
  border: 1px solid white
  text-align: center
  width: 5rem
  font-size: 13px
  vertical-align: middle
  line-height: 25px
  cursor: pointer
