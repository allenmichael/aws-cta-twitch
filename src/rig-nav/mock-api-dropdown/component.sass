.mock-api__dropdown
  display: flex
  width: 10rem
  flex-direction: column
  align-items: center
  font-size: 14px
  color: white
  height: 3rem
  text-align: center
  vertical-align: middle
  margin: 0 1rem
  cursor: pointer

  &:focus
    background-color: #4b367c
    outline-color: #c5b6e2
    outline-width: 0px

.mock-api__menu
  position: absolute
  text-align: left
  margin-top: 3rem
  font-size: 12px
  color: white
  background-color: rgba(32, 29, 44, 0.8)
  width: 10rem
  height: 0
  visibility: hidden
  -webkit-transition: height 0.3s ease
  -moz-transition: height 0.3s ease
  -o-transition: height 0.3s ease
  -ms-transition: height 0.3s ease
  transition: height 0.3s ease, visibility 0.3s
  z-index: 1000
  >ul
    >li
      margin-top: 0.5rem
      >a
        color: white
        text-decoration: none
        >img
          margin-left: .5em

    list-style: none
    margin: 1rem
    padding: 0

.mock-api__menu.open
  height: 3rem
  visibility: visible

.mock-api__header
  display: flex
  flex-direction: row
  align-items: center
  justify-content: space-evenly
  padding: 1rem
  width: 100%
  >img
    margin: auto 0
    height: 10px
    width: 10px



.mock-api__triangle
  color: #fff
  width: 15px
  padding: 0px
  transform: rotate(180deg)
  -webkit-transition: transform 0.3s ease
  -moz-transition: transform 0.3s ease
  -o-transition: transform 0.3s ease
  -ms-transition: transform 0.3s ease
  transition: transform 0.3s ease

.mock-api__triangle.open
  transform: rotate(0deg)
