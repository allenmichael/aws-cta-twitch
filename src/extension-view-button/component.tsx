import * as React from 'react';
import * as line from '../img/rig_line.png';
import * as plus from '../img/plus_icon.png';
import './component.sass'

interface ExtensionViewButtonProps {
  onClick: () => void;
}

type Props = ExtensionViewButtonProps;
export class ExtensionViewButton extends React.Component<Props> {
  public render() {
    return (
      <div className='button-wrapper'>
        <img alt='line' className='background' src={line} />
        <div className="circle-button" onClick={this.props.onClick}>
          <img alt='Add new view' src={plus} />
        </div>
      </div>
    )
  }
}
