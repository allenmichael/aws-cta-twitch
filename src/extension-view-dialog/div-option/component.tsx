import * as React from 'react';
import './component.sass';
import './component.css';

interface DivOptionProps {
  img: string;
  name: string;
  value: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
  checked: boolean;
}
export class DivOption extends React.Component<DivOptionProps> {
  public render() {
    return (
      <div className="extension-container__block">
        <label className="extension-type-label">
          <input type="radio" name="extensionViewType" value={this.props.value} onChange={this.props.onChange} defaultChecked={this.props.checked}/>
          <div className="extension-type-box">
            <img
              className='extension-type-box__img'
              alt={this.props.value}
              src={this.props.img} />
          </div>
          <div className={this.props.checked ? 'extension-type-text dialog-selected': 'extension-type-text'}>{this.props.name}</div>
        </label>
      </div>
    );
  }
}
