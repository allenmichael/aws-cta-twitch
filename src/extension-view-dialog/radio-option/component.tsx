import * as React from 'react';
import './component.sass';

interface RadioOptionProps {
  name: string;
  value: string;
  onChange: (event: React.FormEvent<HTMLInputElement>) => void;
  checked: boolean;
}
type Props = RadioOptionProps & React.HTMLAttributes<HTMLInputElement>
export class RadioOption extends React.Component<Props> {
  public render() {
    return (
      <div className='option-div'>
        <label className="option-label">
            <input type="radio" name={this.props.name} value={this.props.value} onChange={this.props.onChange} defaultChecked={this.props.checked}/>
            <span className={this.props.checked ? 'dialog-selected': null}>{this.props.value}</span>
        </label>
      </div>
    )
  }
}
