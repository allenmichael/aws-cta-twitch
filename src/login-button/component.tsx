import * as React from 'react';
import './component.sass';

export class LoginButton extends React.Component<{}>{
  public render() {
    const rigAuthUrl = 'https://id.twitch.tv/oauth2/authorize?client_id=rtwj27xsv591q1zcnqqfg7suduj4pq&redirect_uri=https://localhost.rig.twitch.tv:3000&response_type=token&scope=user:edit+extensions:edit:catalog';
    return (
        <a href={rigAuthUrl}>
          <button className='login-button'>Sign In</button>
        </a>
    );
  }
}
