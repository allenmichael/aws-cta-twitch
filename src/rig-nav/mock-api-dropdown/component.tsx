import * as React from 'react';
import * as classNames from 'classnames';
import * as greendot from '../../img/greendot.svg';
import * as redx from '../../img/redx.svg';
import * as whitetriangle from '../../img/whitetriangle.svg';

import './component.sass';

export interface ReduxStateProps {
  mockApiEnabled: boolean;
}

export interface ReduxDispatchProps {
  toggleMockApi: () => void;
}

interface State {
  open: boolean;
}

type Props = ReduxStateProps & ReduxDispatchProps;

export class MockApiDropdownComponent extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      open: false
    }
  }

  private handleClick = () => {
    this.setState({
      open: !this.state.open
    });
  }

  public render() {
    const { mockApiEnabled } = this.props;
    const img = mockApiEnabled ? <img src={greendot} alt='enabled' /> : <img src={redx} alt='disabled' />;

    const dropdownClass = classNames({
      'mock-api__menu': true,
      'transition': true,
      'open': this.state.open,
    });

    const triangleClass = classNames({
      'mock-api__triangle': true,
      'open': this.state.open,
    });

    return (
      <div className='mock-api__dropdown' onClick={this.handleClick} tabIndex={0}>
        <div className='mock-api__header'>{img} Mock Responses <img className={triangleClass} src={whitetriangle} alt='open or close' /></div>
        <div className={dropdownClass}>
          <ul>
            <li>
              <div onClick={this.props.toggleMockApi}>{mockApiEnabled ? 'Disable' : 'Enable'}</div>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}
