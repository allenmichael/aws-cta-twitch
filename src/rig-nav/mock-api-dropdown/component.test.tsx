import { setupShallowTest } from '../../tests/enzyme-util/shallow';
import { MockApiDropdownComponent } from './component';

describe('<MockApiDropdownComponent />', () => {
  const setupShallow = setupShallowTest(MockApiDropdownComponent, () => ({
    toggleMockApi: jest.fn()
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls toggleMockApi when enable disable called', () => {
    const { wrapper } = setupShallow();
    wrapper.find('li>div').simulate('click');

    const instance = wrapper.instance() as MockApiDropdownComponent;
    expect(instance.props.toggleMockApi).toHaveBeenCalled();
  });
});
