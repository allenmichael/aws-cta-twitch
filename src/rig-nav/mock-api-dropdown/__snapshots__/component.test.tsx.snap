// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MockApiDropdownComponent /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MockApiDropdownComponent
    toggleMockApi={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="mock-api__header"
>
          <img
                    alt="disabled"
                    src="redx.svg"
          />
           Mock Responses 
          <img
                    alt="open or close"
                    className="mock-api__triangle"
                    src="whitetriangle.svg"
          />
</div>,
        <div
          className="mock-api__menu transition"
>
          <ul>
                    <li>
                              <div
                                        onClick={[Function]}
                              >
                                        Enable
                              </div>
                    </li>
          </ul>
</div>,
      ],
      "className": "mock-api__dropdown",
      "onClick": [Function],
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="disabled"
              src="redx.svg"
/>,
            " Mock Responses ",
            <img
              alt="open or close"
              className="mock-api__triangle"
              src="whitetriangle.svg"
/>,
          ],
          "className": "mock-api__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "disabled",
              "src": "redx.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          " Mock Responses ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "open or close",
              "className": "mock-api__triangle",
              "src": "whitetriangle.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
                        <div
                                    onClick={[Function]}
                        >
                                    Enable
                        </div>
            </li>
</ul>,
          "className": "mock-api__menu transition",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <div
                            onClick={[Function]}
              >
                            Enable
              </div>
</li>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                onClick={[Function]}
>
                Enable
</div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enable",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Enable",
              "type": "div",
            },
            "type": "li",
          },
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mock-api__header"
>
            <img
                        alt="disabled"
                        src="redx.svg"
            />
             Mock Responses 
            <img
                        alt="open or close"
                        className="mock-api__triangle"
                        src="whitetriangle.svg"
            />
</div>,
          <div
            className="mock-api__menu transition"
>
            <ul>
                        <li>
                                    <div
                                                onClick={[Function]}
                                    >
                                                Enable
                                    </div>
                        </li>
            </ul>
</div>,
        ],
        "className": "mock-api__dropdown",
        "onClick": [Function],
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="disabled"
                src="redx.svg"
/>,
              " Mock Responses ",
              <img
                alt="open or close"
                className="mock-api__triangle"
                src="whitetriangle.svg"
/>,
            ],
            "className": "mock-api__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "disabled",
                "src": "redx.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            " Mock Responses ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "open or close",
                "className": "mock-api__triangle",
                "src": "whitetriangle.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <li>
                            <div
                                          onClick={[Function]}
                            >
                                          Enable
                            </div>
              </li>
</ul>,
            "className": "mock-api__menu transition",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                <div
                                onClick={[Function]}
                >
                                Enable
                </div>
</li>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  onClick={[Function]}
>
                  Enable
</div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Enable",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Enable",
                "type": "div",
              },
              "type": "li",
            },
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
